### PR TITLE
fix pubsub pub reading stdin args

### PR DESCRIPTION
**Pull request by whyrusleeping**
License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by whyrusleeping** [comment]
Test failures in bitswap... @Stebalien any idea there?

**Comment by whyrusleeping** [comment]
It looks like we just need a sleep here: https://github.com/ipfs/go-ipfs/blob/master/exchange/bitswap/session_test.go#L315

Cancelling a context is always an async operation, expecting it to affect something else immediately afterwards is a risky bet. In reality, we shouldnt use a context to govern this, we should have an explicit close method, but that will require more refactoring.

File: test/sharness/t0180-pubsub.sh [hunk 1]
```diff
@@ -57,6 +57,36 @@ test_expect_success "wait until echo > wait executed" '
...
+  (
+    ipfsi 2 pubsub sub --enc=ndpayload testTopic | if read line; then
+        echo $line > actual &&
+        echo > wait2
+      fi
+  ) &
+'
+
+test_expect_success "wait until ipfs pubsub sub is ready to do work" '
+  go-sleep 500ms
```
Comment (Kubuxu) [inline test/sharness/t0180-pubsub.sh]: Should we poll here for something (to be sure that pubsub is ready) or not?

Comment (Kubuxu) [reply test/sharness/t0180-pubsub.sh]: I see that we do it in other places so LGTM either way.

Comment (whyrusleeping) [reply test/sharness/t0180-pubsub.sh]: Ideally, but thats pretty difficult to do. For now, this is good enough

**Comment by whyrusleeping** [comment]
@VictorBjelkholm this fixes #4719

**Pull request merged**